core/Updates/5.0.0-b1.php

```diff
@@ -110,4 +117,39 @@ private function requiresUpdatedLogVisitTableIndex()
 
         return true;
     }
+
+    /**
+     * Add new 'pages before column' to log conversions
+     *
+     * @return DbAlias\AddColumns
+     */
+    private function getPagesBeforeAddColumn()
+    {
+        return $this->migration->db->addColumns('log_conversion', ['pageviews_before' => "SMALLINT UNSIGNED DEFAULT NULL"]);
+    }
+
+    private function populatePagesBefore(): void
+    {
+        // Abort if host is *.matomo.cloud or *.innocraft.cloud
+        $instanceId = SettingsPiwik::getPiwikInstanceId();
+        if (strpos($instanceId, '.matomo.cloud') !== false || strpos($instanceId, '.innocraft.cloud') !== false) {
+            return;
+        }
+
+        // Abort if there are more than 10,000 conversions across all sites and goals in the last 48hrs
+        $startDate = Date::factory('yesterday');
+        $endDate = Date::factory('today')->getEndOfDay();
+
+        $sql = 'SELECT COUNT(*) FROM ' . Common::prefixTable('log_conversion') . ' WHERE server_time <= ? AND server_time >= ?';
+
+        $result = Db::fetchOne($sql, [$startDate, $endDate]);
+        if ($result > 10000) {
+            return;
+        }
+
+        // Calculate all conversions for the last 48hrs
+        $pagesBeforeCalculator = new PagesBeforeCalculator();
+        $pagesBeforeCalculator->calculateFor($startDate, $endDate);
```

Comment:
I wonder if we could convert this code to be a custom update instead. This would automatically suggest to run the command when updating manually on console. We did something similar for the token auth migration in Matomo 4. See https://github.com/matomo-org/matomo/blob/5.x-dev/core/Updates/4.0.0-b1.php#L62-L63

Reply:
As discussed, I've removed the check to abort if query if there are 10,000 conversions. 

The migration has been updated to use a `CustomMigration` to run the `core:calculate-conversion-pages` command so that it is shown in the list of dry run commands when using `./console core:update`